### DISCUSSION TITLE
Writing a plugin, I always get None for return_type

### DISCUSSION THREAD

**Discussion by drinking-code** [swc-project/swc#8085] [Q&A] [answered]
I want to write a plugin, but I've run into a problem I can't get around.
When running a visitor over an AST of a file like this: 

```typescript
class MyClass {
  myMethod(): string {
    return 'string';
  }
}
```

where there is a class with a method that has a return type, the return type I get from within the plugin is `None` (see repro [here; run `npm run` to build and run example](https://github.com/drinking-code/swc-return-type-repr).

Am I doing something wrong or is this an actual bug in SWC? The parser recognises the return type; if I look at the AST output by the .parse() function, the return type is properly there. It is only from within the plugin where return_type is `None`

**Comment by magic-akari** [comment, marked as answer]
https://github.com/swc-project/swc/blob/ed648337f1640dfb7d646b1df4047b81b9d6c46d/crates/swc/src/config/mod.rs#L709-L738

The `plugin_transforms` pass is executed after the `typescript` pass. 

https://github.com/swc-project/swc/blob/ed648337f1640dfb7d646b1df4047b81b9d6c46d/crates/swc/src/config/mod.rs#L669-L674

There is an option to only run plugin_transforms. However, I am uncertain of its usability due to the presence of the word `internal_testing` in its name.

**Reply by drinking-code** [reply]
Thanks!
Setting `disableBuiltinTransformsForInternalTesting: true` gives me the type annotations inside the plugin.
(I think that's pretty weird behaviour, but so be it.)

Will see how far I get with this
